File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/common/GenericRepository.java
package ardjomand.leonardo.nutrimeal.data.common;

public interface GenericRepository {

    interface Repository {

        void subscribe();

        void unsubscribe();
    }

    interface Presenter<T> {

        void onItemAdded(T item);

        void onItemChanged(T item);

        void onItemRemoved(String key);
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/common/GenericRepository.kt
package ardjomand.leonardo.nutrimeal.data.common

interface GenericRepository {

    interface Repository {

        fun subscribe()

        fun unsubscribe()
    }

    interface Presenter<T> {

        fun onItemAdded(item: T)

        fun onItemChanged(item: T)

        fun onItemRemoved(key: String)
    }
}
